File: MatlabCodes/Chapter 01/Fig01_02.m
function BivariateNormal
close all;clc;

% Number of points in each direction

n    = 30;

% Parameters

mu_1    = 0;
mu_2    = 0;
sigma_1 = 1;
sigma_2 = 0.5;
[X,Y]   = meshgrid(linspace(-3,3,n),linspace(-3,3,n));

PDF2D =@(rho) mvnpdf([X(:),Y(:)],[mu_1,mu_2],[sigma_1^2,rho*sigma_1*sigma_2;rho*sigma_1*sigma_2,sigma_2^2]);

rho1 = 0;
rho2 = -0.8; 
rho3 = 0.8; 
PDFrho1 = PDF2D(rho1);
PDFrho2 = PDF2D(rho2);
PDFrho3 = PDF2D(rho3);

figure1 = figure('PaperSize',[40.99999864 60]);
set(gcf, 'Position', [400, 500, 1000, 300])
subplot(1,3,1,'Parent',figure1)
mesh(X,Y,reshape(PDFrho1,n,n),'edgecolor', 'k')
xlabel('x')
ylabel('y')
zlabel('PDF')
axis([min(min(X)),max(max(X)),min(min(Y)),max(max(Y)),0,max(max(PDFrho1))])

subplot(1,3,2,'Parent',figure1)
mesh(X,Y,reshape(PDFrho2,n,n),'edgecolor', 'k')
xlabel('x')
ylabel('y')
zlabel('PDF')
axis([min(min(X)),max(max(X)),min(min(Y)),max(max(Y)),0,max(max(PDFrho2))])

subplot(1,3,3,'Parent',figure1)
mesh(X,Y,reshape(PDFrho3,n,n),'edgecolor', 'k')
xlabel('x')
ylabel('y')
zlabel('PDF')
axis([min(min(X)),max(max(X)),min(min(Y)),max(max(Y)),0,max(max(PDFrho3))])

File: PythonCodes/Chapter 01/Fig01_02.py
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import multivariate_normal


def BivariateNormalPDFPlot():
    # Number of points in each direction

    n = 40;

    # Parameters

    mu_1 = 0;
    mu_2 = 0;
    sigma_1 = 1;
    sigma_2 = 0.5;
    rho1 = 0.0
    rho2 = -0.8
    rho3 = 0.8

    # Create a grid and a multivariate normal

    x = np.linspace(-3.0, 3.0, n)
    y = np.linspace(-3.0, 3.0, n)
    X, Y = np.meshgrid(x, y)
    pos = np.empty(X.shape + (2,))
    pos[:, :, 0] = X
    pos[:, :, 1] = Y
    Z = lambda rho: multivariate_normal([mu_1, mu_2], [[sigma_1, rho],
                                                       [rho, sigma_2]])

    # Make a 3D plot- rho = 0.0

    fig = plt.figure(1)
    ax = fig.gca(projection='3d')
    ax.plot_surface(X, Y, Z(rho1).pdf(pos), cmap='viridis', linewidth=0)
    ax.set_xlabel('X axis')
    ax.set_ylabel('Y axis')
    ax.set_zlabel('Z axis')
    plt.show()

    # Make a 3D plot- rho = -0.8

    fig = plt.figure(2)
    ax = fig.gca(projection='3d')
    ax.plot_surface(X, Y, Z(rho2).pdf(pos), cmap='viridis', linewidth=0)
    ax.set_xlabel('X axis')
    ax.set_ylabel('Y axis')
    ax.set_zlabel('Z axis')
    plt.show()

    # Make a 3D plot- rho = 0.8

    fig = plt.figure(3)
    ax = fig.gca(projection='3d')
    ax.plot_surface(X, Y, Z(rho3).pdf(pos), cmap='viridis', linewidth=0)
    ax.set_xlabel('X axis')
    ax.set_ylabel('Y axis')
    ax.set_zlabel('Z axis')
    plt.show()


BivariateNormalPDFPlot()